src/statement_pruner.cpp

```diff
@@ -0,0 +1,101 @@
+#include "statement_pruner.h"
```

Comment:
This is the statement pruner you initially wrote. It is now in its separate file and replaces pruner.h